src/NServiceBus.Transport.SQS/ReducedPayloadSerializerConverter.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace NServiceBus.Transport.SQS
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.Json;
+    using System.Text.Json.Serialization;
+
+    /// <summary>
+    /// This serializer is only used to make sure we are not double storing TimeToBeReceived and ReplyAddress both in
+    /// the headers and on the transport message wrapper to save space on the SQS message body.
+    /// The read method is implemented but never used because only the dispatcher uses it to serialize the transport
+    /// message.
+    /// </summary>
+    sealed class ReducedPayloadSerializerConverter : JsonConverter<TransportMessage>
+    {
+        static readonly JsonSerializerOptions TransportMessageSerializerOptions = new()
+        {
+            TypeInfoResolver = TransportMessageSerializerContext.Default
+        };
+
+        static readonly JsonConverter<Dictionary<string, string>> DefaultDictionaryConverter =
+            (JsonConverter<Dictionary<string, string>>)JsonSerializerOptions.Default.GetConverter(typeof(Dictionary<string, string>));
+        public override TransportMessage Read(ref Utf8JsonReader reader, Type typeToConvert,
+            JsonSerializerOptions options) =>
+            JsonSerializer.Deserialize<TransportMessage>(ref reader, TransportMessageSerializerOptions);
```

Comment:
In practice, we never use the read method. Should I throw here instead?

Reply:
I think so. It's better not to have code we don't need. It can be restored later if we need it for whatever reason.